tenseal/cpp/tensors/ckkstensor.cpp

```diff
@@ -76,7 +76,37 @@ shared_ptr<CKKSTensor> CKKSTensor::power_inplace(unsigned int power) {
 
 shared_ptr<CKKSTensor> CKKSTensor::add_inplace(
     const shared_ptr<CKKSTensor>& to_add) {
-    // TODO
+    // TODO implement broadcasting
+    if (this->_shape != to_add->_shape) {
+        // TODO provide a better message (what are the shapes)
+        throw invalid_argument("Operands doesn't have the same shape");
+    }
+
+    size_t n_jobs = this->tenseal_context()->dispatcher_size();
+
+    auto worker_func = [&](vector<Ciphertext>* dst, vector<Ciphertext>* src,
+                           size_t start, size_t end) -> bool {
+        for (size_t i = start; i < end; i++) {
+            this->tenseal_context()->evaluator->add_inplace((*dst)[i], (*src)[i]);
```

Comment:
I don't think this is thread-safe, accessing dst on the workers.

You should add a mutex for dst here.